### PR TITLE
[API] Update attribute sightings from REST POST

### PR DESCRIPTION
**Pull request by tomking2**
#### What does it do?

This allows you to update attribute-level sightings when POSTing a MISP Event to "updateEvent". This is especially useful in PyMISP when adding sightings to attributes, and updating the entire event to include these sightings. Otherwise, you currently have to manually add in sightings in separate API requests which is long winded.

#### Questions

- [ ] Does it require a DB change?
- [ ] Are you using it in production?
- [ ] Does it require a change in the API (PyMISP for example)?

### CODE REVIEW

File: app/Model/Attribute.php [hunk 1]
```diff
@@ -4077,6 +4077,12 @@ public function editAttribute($attribute, $eventId, $user, $objectId, $log = fal
                 $attribute['distribution'] = 5;
             }
         }
+        if (isset($attribute['Sighting']) && !empty($attribute['Sighting'])) {
+            $this->Sighting = ClassRegistry::init('Sighting');
+            foreach ($attribute['Sighting'] as $s) {
```
Comment (JakubOnderka) [inline app/Model/Attribute.php]: Thanks for contributing. Better to use `Sightings::captureSightings` method.

Example usage:
https://github.com/MISP/MISP/blob/bd98ea937de43b8a311b1f71912ba71d3e674746/app/Model/Attribute.php#L3977-L3979

Comment (tomking2) [reply app/Model/Attribute.php]: Thanks @JakubOnderka, didn't spot that was an available function, will update the code :)

**Comment by tomking2** [comment]
Thanks @JakubOnderka - Updated to use the recommend function. Had to add a further check into that function to pull the Org ID from the user object otherwise the sighting was added without an organisation.

**Comment by JakubOnderka** [comment]
Thanks for contribution!

**Pull request merged**